src/base/src/bson_util.rs

```diff
@@ -34,13 +36,57 @@ pub fn bson_document_into_bytes(doc: &Document) -> Vec<u8> {
     let row_doc = RawDocumentBuf::from_document(doc).unwrap();
     row_doc.into_bytes()
 }
+/// convert bson value to bytes for key comparation
+pub fn bson_into_comparison_bytes(value: &Bson) -> std::result::Result<Option<Vec<u8>>, DB3Error> {
+    let mut data: Vec<u8> = Vec::new();
+    match value {
+        Bson::Null => Ok(None),
+        Bson::Boolean(b) => {
+            data.write_u8(*b as u8)
+                .map_err(|e| DB3Error::DocumentDecodeError(format!("{e}")))?;
+            Ok(Some(data))
+        }
+        Bson::Int64(n) => {
+            if *n >= 0 {
+                data.push(1);
+            } else {
+                data.push(0);
+            }
```

Comment:
```rust
let encoded_data = match  n >= 0 {
    true => n << 1 | 0x01
    false => n << 1 & 0x01
};
data.write_i64::<BigEndian>(encoded_data)
```
                .map_err(|e| DB3Error::DocumentDecodeError(format!("{e}")))?;

Reply:
<img width="958" alt="image" src="https://user-images.githubusercontent.com/6291173/219247661-fc3992c5-6a20-4b1d-9726-44414fa9fb8c.png">

Reply:
OK i will fix the case